### PR TITLE
Add centos9 support to rancher-selinux

### PR DESCRIPTION
**Pull request by andypitcher**
This PR adds  `centos-stream9` (EL9) support to rancher-selinux .

- image: quay.io/centos/centos:stream9

Here is the outcome of the policy compilation and installation, after the build:

```
bash-5.1# cat /etc/*release
CentOS Stream release 9
NAME="CentOS Stream"
VERSION="9"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="9"
PLATFORM_ID="platform:el9"
PRETTY_NAME="CentOS Stream 9"
ANSI_COLOR="0;31"
LOGO="fedora-logo-icon"
CPE_NAME="cpe:/o:centos:centos:9"
HOME_URL="https://centos.org/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"
REDHAT_SUPPORT_PRODUCT="Red Hat Enterprise Linux 9"
REDHAT_SUPPORT_PRODUCT_VERSION="CentOS Stream"
CentOS Stream release 9
CentOS Stream release 9
bash-5.1# ls
CODEOWNERS		   Dockerfile.centos9.dapper		Dockerfile.microos.dapper  README.md
Dockerfile.centos7.dapper  Dockerfile.centos9.dapper2218901142	LICENSE			   policy
Dockerfile.centos8.dapper  Dockerfile.centos9.dapper2819780759	Makefile
bash-5.1# cd policy/
bash-5.1# ls
centos7  centos8  centos9  microos
bash-5.1# cd centos9/
bash-5.1# ls
rancher-selinux.spec  rancher.fc  rancher.te  scripts
bash-5.1# make -f /usr/share/selinux/devel/Makefile rancher.pp
Compiling targeted rancher module
Creating targeted rancher.pp policy package
rm tmp/rancher.mod.fc tmp/rancher.mod
bash-5.1# semodule -i rancher.pp
libsemanage.semanage_rename: WARNING: rename(/var/lib/selinux/targeted/active, /var/lib/selinux/targeted/previous) failed: Invalid cross-device link, fall back to non-atomic semanage_copy_dir_flags()
```

Tracking issue in Rancher https://github.com/rancher/rancher/issues/43146.

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -174,6 +174,96 @@ steps:
...
+
+- name: GitHub Release EL9
+  image: ibuildthecloud/github-release:v0.0.1
+  settings:
+    api_key:
+      from_secret: github_token
+    prerelease: true
+    checksum:
+    - sha256
+    checksum_file: CHECKSUMsum-centos8-noarch.txt
```
Comment (macedogm) [inline .drone.yml:254]:
```suggestion
    checksum_file: CHECKSUMsum-centos9-noarch.txt
```

File: .drone.yml [hunk 2]
```diff
@@ -174,6 +174,96 @@ steps:
...
+    PRIVATE_KEY:
+      from_secret: private_key
+    PRIVATE_KEY_PASS_PHRASE:
+      from_secret: private_key_pass_phrase
+    TESTING_PRIVATE_KEY:
+      from_secret: testing_private_key
+    TESTING_PRIVATE_KEY_PASS_PHRASE:
+      from_secret: testing_private_key_pass_phrase
+  commands:
+  - policy/centos8/scripts/sign
```
Comment (macedogm) [inline .drone.yml:210]:
```suggestion
  - policy/centos9/scripts/sign
```

**Review by macedogm — CHANGES_REQUESTED** [review]

File: .drone.yml (hunk 1 above)
Comment (andypitcher) [reply .drone.yml]: Nice catch, I've updated the commit with both of your changes - thanks.

**Review by macedogm — APPROVED** [review]

**Comment by andypitcher** [comment]
rancher-selinux test on Centos9 (RHEL9) was successful.

Environment and versions used:

- Server: Centos Stream 9 x86_64
- Cluster: rke1-v1.26.8 (standalone with etcd/controplane/worker)
- kubectl: v1.28.2
- helm: v3.13.1
- rancher: v2.7.6

Generate rancher-selinux rpm:

1. Generate the rancher-linux rpm (use the centos-stream9 branch), and copy the generated files locally from the container (will use it later to install it on the Centos9 server).
```
> podman run --rm -it -v $(pwd):/code quay.io/centos/centos:stream9
# yum install -y epel-release && yum -y install container-selinux selinux-policy-devel yum-utils rpm-build git
# ./policy/centos9/scripts/build
```

Procedure to test rancher-selinux rpm:

1. Spin up a Centos9 instance and install the requirements (rke, kubectl, helm, docker and SELinux packages)
  ```
    1  yum update -y
    2  yum install -y epel-release && yum -y install container-selinux selinux-policy-devel setools
    3  curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl"
    5  mv kubectl /usr/bin/
    6  chmod +x /usr/bin/kubectl
    8  yum install -y yum-utils
    9  yum-config-manager --add-repo https://download.docker.com/linux/centos/docker-ce.repo
   10  yum install docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin
   12  curl -LO https://github.com/rancher/rke/releases/download/v1.4.10/rke_linux-amd64
   14  chmod +x rke_linux-amd64
   15  mv rke_linux-amd64 /usr/bin/rke
```
3. Create a RKE1 cluster, use the [RKE install minimum conf](https://rke.docs.rancher.com/installation#creating-the-cluster-configuration-file)
4.

**Comment by macedogm** [comment]
@andypitcher I believe we can merge this PR now and generate a testing RPM, wdyt?

Thanks for adding the [comment](https://github.com/rancher/rancher-selinux/pull/20#issuecomment-1766488815) above. I believe that it should be pasted as is in https://github.com/rancher/rancher/issues/43146, as it will help QA when testing this.

**Pull request merged**